.gitignore

```diff
@@ -11,7 +11,6 @@ cmake-build*
 
 # npm
 node_modules/
-package-lock.json
```

Comment:
Removing the `package-lock.json` will sadly be required in order for users to build and use this package. As the buildNpmPackage depends on it for vendoring it's dependencies. We can't rely on cmake to install the NPM deps since it's a offline build for repeatability. It's this a show stopper?

Reply:
> It's this a show stopper?

Yes, I think so. I don't want dependabot updating every sub dependency.

As an alternative, how about creating a secondary repo where you can host this flake (and pre-built version if nix supports that)? Then open a PR adding something to the docs about it. We have a section for third party packages. https://github.com/LizardByte/Sunshine/blob/d1a635809ac8cf7b16913479648d067159119e97/docs/source/about/third_party_packages.rst?plain=1#L20

I'd still be happy to review any build spec or CI stuff you come up with, whether in this repo or your own.

Reply:
I think that any sane NodeJS setup should include the dependency lock file `package-lock.json` in version control. 
Its the only way to make sure developers and CI will install the exact same dependencies (even 5 years from now). 
Generating it at build and/or at CI makes no sense.
dependabot can be configured separately https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file

Reply:
That's pointless for our project. We hard pin our required dependencies... I'm not dealing with dependabot updating dependencies of dependencies. Lock files are good for libraries that soft pin dependencies. I already know how it works with lock files, and it's incredibly annoying. Sometimes opening multiple PRs for the same dependency in the two different files. No thanks.

Reply:
Even if the main dependencies are locked a different install could bring in some different and potentialy broken dependencies of dependencies.